src/Checker.ts

```diff
@@ -13,35 +13,42 @@ export const checkerToGuard: <FROM, TO extends FROM, N extends FROM = FROM>(
 	);
 
 function getRawGuard<FROM, TO extends FROM>(checker: Checker<FROM>): ReasonGuard<FROM, TO> {
-	return (input, e = [], c = [], context = []): input is TO => {
+	return (input, errors, confirmations, context = []): input is TO => {
+		let result: string | ErrorLike;
 		try {
-			c.push(checker(input, context));
-			return true;
-		} catch (err: any) {
-			if (err instanceof CompositeError) {
-				e.push(...err.errors);
-			} else {
-				e.push(err);
+			result = checker(input, context);
+			if (typeof result === 'string') {
+				confirmations?.push(result);
+				return true;
 			}
-			return false;
+		} catch (err: any) {
+			result = err;
```

Comment:
since we're not throwing errors anymore, do we need to still convert caught errors into a reason-guard result? might that be hiding bugs?

Reply:
this is primarily backwards compatibility -- in case any app code that pulls this in is still using `throw`.

but it's also defensive code -- validation shouldn't throw errors, it should return validation errors. So if an inner checker/guard throws, we should treat that as a validation error, same as before.

Reply:
👍